src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -771,7 +780,7 @@ private Task FillAsync(CancellationToken cancellationToken)
                 int bytesRead = t.GetAwaiter().GetResult();
                 if (NetEventSource.IsEnabled) Trace($"Received {bytesRead} bytes.");
                 _readLength += bytesRead;
-                return Task.CompletedTask;
+                return Task.FromResult(bytesRead > 0);
```

Comment:
FromResult doesn't return cached tasks (at least not currently).  If we keep this returning `Task<bool>` rather than throwing on EOF per my earlier question, we should cache this task:
```C#
private static readonly Task<bool> s_trueTask = Task.FromResult(true);
private static readonly Task<bool> s_falseTask = Task.FromResult(false);
...
return bytesRead > 0 ? s_trueTask : s_falseTask;
```
There's an active discussion about whether FromResult should be changed to pull from the same task cache that async methods do for synchronously completing methods, and if that changes this could be undone, but for now, let's manually cache.